### PR TITLE
Added support for typescript declarations

### PR DESCRIPTION
**Pull request by anikesh-butler**
Added typescript declarations support with standard conversions followed by the community. After this update, the `lib` will serve the purpose of input source for npm consumers. And dist will server the purpose for singleton JS file for non-npm environments. 

To build run
```bash
yarn build
```

This will generate the `dist` and `lib` directory. After this, it's ready to be published into the npm repository.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const myDocument = {
 };
 
 // Step 3: Fetch data about your document from Butler
-const fetchDocumentData = async ({ modelId, documentId }) => {
+const fetchDocumentData = async (modelId, documentId) => {
```
Comment (JacobGreenButler) [inline index.js:19]: can we add these changes to the README.md file as well?

File: index.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ const onSaveCallback = (docInfo) => {
 
 // This function will inject the Butler Document Labeler into the
 // div element you specified earlier with the fetched document data
-const initializeDocLabeler = ({ modelId, documentId }) => {
+const initializeDocLabeler = async ({ modelId, documentId }) => {
```
Comment (JacobGreenButler) [inline index.js:48]: ditto about adding to readme

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 node_modules
-dist
\ No newline at end of file
+dist
+lib
```
Comment (JacobGreenButler) [inline .gitignore:3]: what is lib?

**Comment by JacobGreenButler** [comment]
> _ No description provided. _

Can we add a description for this MR? Additionally, can we call out the approach we're using here for the typing declarations?

File: package.json [hunk 4]
```diff
@@ -1,19 +1,22 @@
 {
   "name": "butler-blocks",
-  "version": "1.0.2",
+  "version": "1.0.3",
   "description": "Butler Labs Inc.'s Library of embeddable components and apis to power AI for developers",
-  "main": "dist/butlerBlocks.js",
-  "module": "dist/butlerBlocks.js",
+  "main": "lib/butlerBlocks.js",
+  "module": "lib/butlerBlocks.js",
+  "types": "lib/butlerBlocks.d.ts",
```
Comment (JacobGreenButler) [inline package.json:7]: why did we decide to update these?

File: package.json [hunk 5]
```diff
@@ -37,6 +40,7 @@
     "jest": "^28.1.3",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "shx": "^0.3.4",
```
Comment (JacobGreenButler) [inline package.json:43]: what is shx?

File: package.json [hunk 6]
```diff
@@ -1,19 +1,22 @@
...
   "files": [
+    "lib",
     "dist",
     "README.md"
   ],
   "scripts": {
     "test": "jest",
     "start": "webpack-dev-server --open --mode development --hot",
-    "build": "webpack --mode production",
+    "build": "yarn run clean && tsc && webpack --mode production",
```
Comment (JacobGreenButler) [inline package.json:16]: Do we need yarn and webpack?

**Review by JacobGreenButler — COMMENTED** [review]
Seems like this MR needs a bit more clarification on the actual approach taken

File: index.js (hunk 1 above)
Comment (anikesh-butler) [reply index.js:19]: done

File: index.js (hunk 2 above)
Comment (anikesh-butler) [reply index.js:48]: done

File: .gitignore (hunk 3 above)
Comment (anikesh-butler) [reply .gitignore:3]: checkout the description for details.

File: package.json (hunk 4 above)
Comment (anikesh-butler) [reply package.json:7]: it's as per conventions. like `lib` will be consumed by the npm and it'll contain the typescript compiled code with type declarations for typescript projects.
As `dist` is only for distribution purposes through cdn.

File: package.json (hunk 5 above)
Comment (anikesh-butler) [reply package.json:43]: It's cross-platform shell.

File: package.json (hunk 6 above)
Comment (anikesh-butler) [reply package.json:16]: yes, just to clear the existing generated code for sake of simplicity.

File: package.json (hunk 4 above)
Comment (anikesh-butler) [reply package.json:7]: Can be seen in any well-known package inside node_modules directory. like @material-ui

**Review by JacobGreenButler — APPROVED** [review]

**Pull request merged**